tests/test_zarr_write.py

```diff
@@ -0,0 +1,150 @@
+    h5f.attrs['attr_list_float'] = [1.1, 2.2, 3.3]
+    h5f.attrs['attr_list_bool'] = [True, False, True]
+    with pytest.raises(Exception):
+        h5f.attrs['attr_list_mixed'] = [1, 2.2, 'c', True]
+    h5f.attrs['2d_array'] = np.array([[1, 2], [3, 4]])
+    h5f.create_dataset('dset_int8', data=np.array([1, 2, 3], dtype=np.int8))
+    h5f.create_dataset('dset_int16', data=np.array([1, 2, 3], dtype=np.int16))
+    h5f.create_dataset('dset_int32', data=np.array([1, 2, 3], dtype=np.int32))
+    h5f.create_dataset('dset_int64', data=np.array([1, 2, 3], dtype=np.int64))
+    h5f.create_dataset('dset_uint8', data=np.array([1, 2, 3], dtype=np.uint8))
+    h5f.create_dataset('dset_uint16', data=np.array([1, 2, 3], dtype=np.uint16))
+    h5f.create_dataset('dset_uint32', data=np.array([1, 2, 3], dtype=np.uint32))
+    h5f.create_dataset('dset_uint64', data=np.array([1, 2, 3], dtype=np.uint64))
+    h5f.create_dataset('dset_float32', data=np.array([1, 2, 3], dtype=np.float32))
+    h5f.create_dataset('dset_float64', data=np.array([1, 2, 3], dtype=np.float64))
+    h5f.create_dataset('dset_bool', data=np.array([True, False, True], dtype=np.bool_))
+
+    group1 = h5f.create_group('group1')
+    group1.attrs['attr_str'] = 'hello'
+    group1.attrs['attr_int'] = 42
+    group1.create_dataset('dset_with_nan', data=np.array([1, np.nan, 3], dtype=np.float64))
+    group1.create_dataset('dset_with_inf', data=np.array([np.inf, 6, -np.inf], dtype=np.float64))
+
+    compound_dtype = np.dtype([('x', np.int32), ('y', np.float64)])
+    group1.create_dataset('dset_compound', data=np.array([(1, 2.2), (3, 4.4)], dtype=compound_dtype))
+
+    group_to_delete = h5f.create_group('group_to_delete')
+    group_to_delete.attrs['attr_str'] = 'hello'
+    group_to_delete.attrs['attr_int'] = 42
+    group_to_delete.create_dataset('dset_to_delete', data=np.array([1, 2, 3], dtype=np.int8))
+    del h5f['group_to_delete']
+
+    another_group_to_delete = group1.create_group('another_group_to_delete')
+    another_group_to_delete.attrs['attr_str'] = 'hello'
+    another_group_to_delete.attrs['attr_int'] = 42
+    another_group_to_delete.create_dataset('dset_to_delete', data=np.array([1, 2, 3], dtype=np.int8))
+    del group1['another_group_to_delete']
+
+    yet_another_group_to_delete = group1.create_group('yet_another_group_to_delete')
+    yet_another_group_to_delete.attrs['attr_str'] = 'hello'
+    yet_another_group_to_delete.attrs['attr_int'] = 42
+    yet_another_group_to_delete.create_dataset('dset_to_delete', data=np.array([1, 2, 3], dtype=np.int8))
+    del h5f['group1/yet_another_group_to_delete']
+
+
+def compare_example_h5_data(h5f: h5py.File, tmpdir: str):
+    with h5py.File(f'{tmpdir}/for_comparison.h5', 'w') as h5f2:
+        write_example_h5_data(h5f2)
+    with h5py.File(f'{tmpdir}/for_comparison.h5', 'r') as h5f2:
+        _assert_groups_equal(h5f, h5f2)
+
+
+def _assert_groups_equal(h5f: h5py.Group, h5f2: h5py.Group):
```

Comment:
I suggest moving this and all the methods below to a module `tests/utils.py`, removing the leading underscore, and importing them into the tests that use them. Currently `_arrays_are_equal` is copied to `test_copy.py` and `test_core.py`

Reply:
done